### PR TITLE
Upgrade to Leaflet 1.0

### PR DESCRIPTION
**Pull request by RickMohr**
Adjust for Leaflet map pane`z-index` changes (now 200-700 rather than small integers)
- Search box on `create` page now at 900
- `.map` is the root Leaflet map element. Add `z-index: 1` (overriding Leaflet's 400) so dropdowns above it will overlay it correctly.
- Adjust z-index and position of modeling controls / info displays (they should really all be Leaflet controls, but that's outside the scope of this PR)

Put "searched boundary" polygon between base map and tiles in the map layer stacking order
* Add `base-map` pane to contain base map layer
* Add `searched-boundary` pane to contain searched boundary polygon
* Simplify code creating ESRI base map (workarounds no longer needed)

Deal with all Leaflet plugins
- `npm`
  - `esri-leaflet`
    - Still works at current version (~1.0.2). Latest is 2.0.7, but no reason to update since we don't actually use it. We could now reduce its download size, see https://github.com/OpenTreeMap/otm-core/issues/2966
  - `leaflet-pip`
    - Had version ~0.1.1
    - Now fetching from `npm` and updated to latest, 1.0.0
- `vendor`
  - `leaflet.draw.js`
    - Had version 0.2.2
    - Now fetching from `npm` and updated to latest, 0.4.9
- `shim`
  - `leaflet.bing.js`
    - Still works fine, though I'm not sure where ours came from
    - There are also these:
      - https://github.com/mertemin/leaflet-bing
      - https://github.com/gmaclennan/leaflet-bing-layer
  - `leaflet-editable-polyline.js`
    - Abandoned
    - Crashes with Leaflet 1.0
    - Removed in favor of `leaflet-draw`
  - Google Maps tiles
    - Were using `leaflet.google.js` (https://gist.github.com/crofty/2197042)
    - Switched to `Leaflet.GridLayer.GoogleMutant` from `npm`
      - Solves layer sync problem -- tree dots and basemap now pan together!!
      - Doesn't work on IE10 (but that's only 0.8% of market; already losing 4.3% IE9)
  - `leaflet.utfgrid.js`
    - replaced shim with latest version 0.2.1

Also:
  - Green infrastructure polygon editing now uses `leaflet-draw` (whose vertex boxes are BIG).
  - Add "Terrain" option, now available on Google base maps
  - `maxNativeZoom` is honored by Leaflet 1.0, so the canopy layer on `patreemap` no longer disappears at high zoom levels
  - Tweak margin on tiny "Leaflet" button at lower right of map so it doesn't overlap Google Map attribution
  - Removed `fixZoomLayerSwitch` workaround

Testing - verify that all Leaflet maps work right
  - Main map page
      - Search dropdowns (regular and advanced) correctly cover the map
      - You can still add and move a tree and bioswale
      - `patreemap` canopy layers stack and toggle correctly; canopy shows at max zoom
      - The polygon shown after searching by zip code is correctly beneath the tree dots
      - Tree dots and basemap now pan together
  - Mini map on plot detail page
  - "Create" step 3
  - Modeling
      - "distribution" polygons work right -- add, select, view area, edit

Connects #2812
Connects #776
Connects #578

### CODE REVIEW

File: opentreemap/treemap/templates/treemap/partials/map_add_resource.html [hunk 1]
```diff
@@ -52,7 +52,7 @@
                 <a href="javascript:;" class="close cancelBtn small visible-xs-block">×</a>
             </div>
             <div class="add-step-content">
-                <div class="alert alert-info">Move the dots on the blue polygon to define the borders of your {{ term.Resource.singular.lower }}.
+                <div class="alert alert-info">"Move the squares on the blue polygon to define the borders of your {{ term.Resource.singular.lower }}.
```
Comment (jwalgran) [inline opentreemap/treemap/templates/treemap/partials/map_add_resource.html]: Looks like it

File: assets/css/vendor/leaflet.css [hunk 2]
```diff
@@ -240,55 +298,38 @@
...
 .leaflet-control-layers {
-	box-shadow: 0 1px 7px rgba(0,0,0,0.4);
-	background: #f8f8f9;
-	-webkit-border-radius: 5px;
-	        border-radius: 5px;
+	box-shadow: 0 1px 5px rgba(0,0,0,0.4);
+	background: #fff;
+	border-radius: 5px;
 	}
 .leaflet-control-layers-toggle {
```
Comment (RickMohr) [inline assets/css/vendor/leaflet.css]: I don't see this issue -- try `scripts/manage.sh collectstatic`.

(Note that without branding colors webpack bundles the images directly but with branding colors the dynamically-compiled SASS refers to the static image location.)

Comment (jwalgran) [inline assets/css/vendor/leaflet.css]: Running `collectstatic` resolved the issue. Thanks.

File: opentreemap/opentreemap/settings/default_settings.py [hunk 3]
```diff
@@ -406,7 +406,7 @@
 
 # We have... issues on IE9 right now
 # Disable it on production, but enable it in Debug mode
-IE_VERSION_MINIMUM = 9 if DEBUG else 10
+IE_VERSION_MINIMUM = 9 if DEBUG else 11
```
Comment (jwalgran) [inline opentreemap/opentreemap/settings/default_settings.py]: I tested this with BrowserStack and saw the redirect working in IE 10.

We have a hard-coded version "10" in our `unsupported.html` template that we should change as well.

https://github.com/OpenTreeMap/otm-core/blob/77f662725fc4bd3ff15e8a6817f9799757db514f/opentreemap/treemap/templates/treemap/unsupported.html

Comment (RickMohr) [inline opentreemap/opentreemap/settings/default_settings.py]: Thanks for remembering! Fixed.

**Pull request merged**